### PR TITLE
Update instructions for cross compilation

### LINKED ISSUE #26: Build instructions outdated
I think the readme does not show the correct way to build the container. This is a normal buildx capable Dockerfile.

Also I think the build shell script is sort of redundant and not used anywhere

Copy from mailinglist

```
We found a maybe a minor issue in the Dockerfile when building it for arm64 machines.

Background:
We are creating a POC using the vehicle-app-cpp-template.
Everything worked great when we built the docker images in our host machine (amd64) and deployed it in ubuntu test machine (amd64).
However, when we built the docker image for arm64 and still using our host machine (amd64) and to be deployed in our edge devices (arm64).
The final image is still in amd64.

This is how docker build the apps:
$ cd kuksa-can-provider
$ docker build -f Dockerfile --progress=plain --build-arg TARGETPLATFORM=linux/arm64 -t localhost:12345/feeder-can:0.4.3 .

After building it we docker image inspect the docker image.: 
$ docker image inspect localhost:12345/feeder-can:0.4.3 | grep Arch
$ "Architecture": "amd64",

We fixed it by modifying cd kuksa-can-provider/Dockerfile.
# FROM gcr.io/distroless/base-debian12
FROM --platform=$TARGETPLATFORM gcr.io/distroless/base-debian12
and

After this minor fix. We can now successfully deploy it with our edge devices (arm64).

Does our Dockerfile changes is the correct way to go forward or do you recommend another way to build docker(arm64) images using host(amd64)?
```

### PR DESCRIPTION
**Pull request by erikbosch**
Also removing unused file

Fixes #26

### CODE REVIEW

File: doc/docker.md [hunk 1]
```diff
@@ -14,6 +14,18 @@ docker run  --net=host -e LOG_LEVEL=INFO can-provider:latest --server-type kuksa
 docker run  --net=host -e LOG_LEVEL=INFO can-provider:latest --server-type kuksa_val_server
 ```
 
+## Building for a different architecture
+
+You can build for a different architecture by using the `--platform` argument.
+You might need to install `qemu-user-static` depending on your distribution.
+
+```console
+docker build -f Dockerfile --progress=plain --build-arg --platform=linux/arm64 -t can-provider:latest .
```
Comment (SebastianSchildt) [inline doc/docker.md:23]: I am not so sure if, this does "all the necessary things", I would prefer we just give the buildx version

Comment (erikbosch) [reply doc/docker.md:23]: As I understand from https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/ and https://github.com/docker/cli/pull/3314 the two commands are today equal.

So then it becomes more a question of which syntax we want to promote :-)

I tested building with both, I however only verified that it reported the right architecture.

Comment (SebastianSchildt) [reply doc/docker.md:23]: If it is equivalent, what is `--build-arg` doing?  In any case "buildx" might then be more compatible, and also working on docker(-cli?)  < 23.00.

Comment (erikbosch) [reply doc/docker.md:23]: That is a type, but I can remove the first alternative

**Review by SebastianSchildt — APPROVED** [review]

**Pull request merged**